gensim/models/nmf.py

```diff
@@ -557,30 +557,35 @@ def update(self, corpus, chunksize=None, passes=None, eval_every=None):
         if eval_every is None:
             eval_every = self.eval_every
 
+        lencorpus = np.inf
+        evalafter = "?"
+
         try:
             if isinstance(corpus, scipy.sparse.csc.csc_matrix):
                 lencorpus = corpus.shape[1]
             else:
                 lencorpus = len(corpus)
+
+            if chunksize is None:
+                chunksize = min(lencorpus, self.chunksize)
+
+            evalafter = min(lencorpus, (eval_every or 0) * chunksize)
         except Exception:
-            logger.warning("input corpus stream has no len(); counting documents")
-            lencorpus = sum(1 for _ in corpus)
+            logger.warning("input corpus stream has no len()")
```

Comment:
No `len()` is a perfectly valid input state, the warning is too severe. Maybe INFO?

Reply:
Alright